### PR TITLE
fix(#6671): fix possible stack buffer overflow in gen-s-parser.inc

### PR DESCRIPTION
**Pull request by mtb0x1**
The stack buffer overflow is occurring because memcpy(buf, op.data(), op.size()); can write up to op.size() bytes into buf, but buf is only 33 bytes long. If op.size() is greater than 33, this will result in a buffer overflow.

### CODE REVIEW

File: scripts/gen-s-parser.py [hunk 1]
```diff
@@ -708,6 +708,10 @@ def instruction_parser():
 
     printer.print_line("auto op = *keyword;")
     printer.print_line("char buf[{}] = {{}};".format(inst_length + 1))
+    printer.print_line("// Ensure we do not copy more than the buffer can hold")
+    printer.print_line("if (op.size() >= sizeof(buf)) {")
+    printer.print_line("	goto parse_error;")
```
Comment (kripken) [inline scripts/gen-s-parser.py:713]:
```suggestion
    printer.print_line("  goto parse_error;")
```
(spaces instead of tabs)

**Review by tlively — APPROVED** [review]
Thanks!

**Comment by tlively** [comment]
Can you try emitting `// NOLINTBEGIN` and `// NOLINTEND` comments around the generated code to solve the clang-tidy error on CI?

**Comment by mtb0x1** [comment]
@tlively I added your suggestion but the lint runner is still failing, and it's failing on first line of gen-s-parser.inc ... for some reason.

**Pull request merged**